src/cpp/include/openvino/genai/whisper_pipeline.hpp

```diff
@@ -12,37 +12,69 @@
-        OPENVINO_THROW("This method is deprecated and will be removed in 2026.0.0 release. Please, override write_chunk() insted.");
+    OPENVINO_DEPRECATED("ChunkStreamerBase is deprecated and will be removed in 2026.0.0 "
+                        "release. Use StreamerBase instead.")
+    OPENVINO_GENAI_EXPORTS virtual bool put_chunk(std::vector<int64_t> tokens) {
+        OPENVINO_THROW(
+            "ChunkStreamerBase is deprecated and will be removed in 2026.0.0 release. Use StreamerBase instead.");
         return true;
-    }
+    };
 
-    /// @brief write_chunk is called every time new token chunk is generated
-    /// @return StreamingStatus flag to indicate whether generation should be stopped
-    virtual StreamingStatus write_chunk(std::vector<int64_t> tokens) {
+    /// @brief put is called every time new token is decoded. Deprecated. Please, use write instead.
+    /// @return bool flag to indicate whether generation should be stopped, if return true generation stops
+    OPENVINO_DEPRECATED(
+        "ChunkStreamerBase is deprecated and will be removed in 2026.0.0 release. Use StreamerBase instead.")
+    OPENVINO_GENAI_EXPORTS virtual bool put(int64_t token) override {
+        OPENVINO_THROW(
+            "ChunkStreamerBase is deprecated and will be removed in 2026.0.0 release. Use StreamerBase instead.");
+        return true;
+    };
+
+    /// @brief write is called every time new vector of tokens is decoded, in case of assisting or prompt lookup
+    /// decoding
+    /// @return StreamingStatus flag to indicate whether generation should be countinue to run or stopped or cancelled
+    OPENVINO_DEPRECATED(
+        "ChunkStreamerBase is deprecated and will be removed in 2026.0.0 release. Use StreamerBase instead.")
+    OPENVINO_GENAI_EXPORTS virtual StreamingStatus write(const std::vector<int64_t>& tokens) override {
+        OPENVINO_SUPPRESS_DEPRECATED_START
         return put_chunk(tokens) ? StreamingStatus::STOP : StreamingStatus::RUNNING;
-    }
+        OPENVINO_SUPPRESS_DEPRECATED_END
+    };
+
+    /// @brief write is called every time new token is decoded
+    /// @return StreamingStatus flag to indicate whether generation should be countinue to run or stopped or cancelled
+    OPENVINO_DEPRECATED(
+        "ChunkStreamerBase is deprecated and will be removed in 2026.0.0 release. Use StreamerBase instead.")
+    OPENVINO_GENAI_EXPORTS virtual StreamingStatus write(int64_t token) override {
+        OPENVINO_SUPPRESS_DEPRECATED_START
+        return put(token) ? StreamingStatus::STOP : StreamingStatus::RUNNING;
+        OPENVINO_SUPPRESS_DEPRECATED_END
+    };
+
+    /// @brief end is called at the end of generation. It can be used to flush cache if your own streamer has one
+    OPENVINO_DEPRECATED(
+        "ChunkStreamerBase is deprecated and will be removed in 2026.0.0 release. Use StreamerBase instead.")
+    OPENVINO_GENAI_EXPORTS virtual void end() override = 0;
```

Comment:
```suggestion
    void end() override = 0;
```
you don't need to export each method as class is already exported.

Reply:
accidental change, removed, thanks